src/models/BaseModel.py

```diff
@@ -0,0 +1,70 @@
+import math
+
+import abc
+import pandas as pd
+import sklearn
+
+from cleaning_functions import pandas_vector_to_list
+
+
+class BaseModel:
+    def __init__(self, train_raw_data_csv_file_name='../data/Train_rev1.csv',
+                 train_normalized_location_file_name='../data/train_normalised_location.csv',
+                 train_test_ratio=0.8, load_location=True):
+        # TODO: make this static with single load
+        self.raw_data = pd.read_csv(train_raw_data_csv_file_name)
+
+        # split original training dataset into train and test data
+        # since we most likely wont get the test salary
+        self.full_dataset_size = len(self.raw_data)
```

Comment:
we can do this directly using functionality already built into scikit_learn - function is `sklearn.model_selection.train_test_split`. I'm not 100% sure here, but it should also work with pandas dataframes directly - so we can avoid doing this...

```
        self.train_contract_type_feature = self.train_raw_data[['ContractType']]
        self.train_contract_time_feature = self.train_raw_data[['ContractTime']]
        self.train_category_feature = self.train_raw_data[['ContractTime']]
        self.train_company_feature = self.train_raw_data[['Company']]
        self.train_source_name_feature = self.train_raw_data[['SourceName']]
        self.train_location_raw_feature = self.train_raw_data[['LocationRaw']]
        self.train_salary_normalized = self.train_raw_data[['SalaryNormalized']]
```